src/bridge/cockpitpipechannel.c

```diff
@@ -384,6 +407,10 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
       else
         self->pipe = cockpit_pipe_spawn ((const gchar **)argv, (const gchar **)env, dir, flags);
     }
+  else if (internal && lookup_internal_fd (internal, &fd))
```

Comment:
So the following line claims the fd, however lookup_internal_fd doesn't steal it from the hashtable or otherwise transfer ownership.

I imagine that such fd channels are open-once channels, after which the identifier becomes invalid? Or did you imagine them to be able to open many channels to the same fd? This is obviously a corner case, but either way needs to be made consistent.